---
title: "General"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{General}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---
<!-- Website built with pkgdown::build_site()  -->
```{r setup, echo=FALSE}
library(VCFtoGWAS)
```
<head>
<meta name="viewport" content="width=device-width, initial-scale=1">
<style>
body {font-family: Arial;}

/* Style the tab */
.tab {
  overflow: hidden;
  border: 1px solid #ccc;
  background-color: #f1f1f1;
}

/* Style the buttons inside the tab */
.tab button {
  background-color: inherit;
  float: left;
  border: none;
  outline: none;
  cursor: pointer;
  padding: 14px 16px;
  transition: 0.3s;
  font-size: 17px;
}

/* Change background color of buttons on hover */
.tab button:hover {
  background-color: #ddd;
}

/* Create an active/current tablink class */
.tab button.active {
  background-color: #ccc;
}

/* Style the tab content */
.tabcontent {
  display: none;
  padding: 6px 12px;
  -webkit-animation: fadeEffect 1s;
  animation: fadeEffect 1s;
}


@-webkit-keyframes fadeEffect {
  from {opacity: 0;}
  to {opacity: 1;}
}

@keyframes fadeEffect {
  from {opacity: 0;}
  to {opacity: 1;}
}
</style>
</head>

<body>
<div class="tab">
  <button class="tablinks" onclick="openCity(event, 'Motivation')">Motivation</button>
  <button class="tablinks" onclick="openCity(event, 'Usage')">Usage</button>
  <button class="tablinks" onclick="openCity(event, 'Installing the package')">Installing</button>
  <button class="tablinks" onclick="openCity(event, 'Output structure')">Structure</button>
  <button class="tablinks" onclick="openCity(event, 'VCF info')">VCF</button>
</div>

<!-- ### Motivation -->
<div id="Motivation" class="tabcontent">
<h3>Motivation</h3>
#### *GWAS*

The aim of Genome‐wide association studies (GWAS) is to identify single nucleotide polymorphisms (SNPs) of which the allele frequencies vary systematically as a function of phenotypic trait values.

Identification of trait‐associated SNPs may subsequently reveal new insights into the biological mechanisms underlying these phenotypes.

The main package used here for GWAS is "[statgenGWAS](https://cran.r-project.org/web/packages/statgenGWAS/vignettes/GWAS.html)"

#### *VCF*

VCF files are big genetic files and working with them might be complicated.

The main package used here for loading the vcf file is "[vcfR](https://cran.r-project.org/web/packages/vcfR/vignettes/)"

</div>


<!-- ### Usage -->
<div id="Usage" class="tabcontent">
<h3>Usage</h3>
If a server with stronger computation is available, it is required to run most of the functions on it (especially chapter 1).\
For now, the R markdown files that show how to use the package step by step (chapter 1 and chapter 2) aren't publicly available. You can send an email [the maintainer]() of the package

[For Weizmann institute users: The package is recommended to be executed on the WEXAC R server. All the required packages are installed on R 4.1.1 so this is recommended. As for plotting, there are some issues with the WEXAC server so save the files and run the plots and pdf creations from your computer (unless you know how to save it)]{style="font-size: 10pt"}

</div>


<!-- ### Installing the package -->
<div id="Installing the package" class="tabcontent">
<h3>Installing the package</h3>
If a server with stronger computation is available, it is required to run most of the functions on it (especially chapter 1).\
For now, the R markdown files that show how to use the package step by step (chapter 1 and chapter 2) aren't publicly available. You can send an email [the maintainer]() of the package

[For Weizmann institute users: The package is recommended to be executed on the WEXAC R server. All the required packages are installed on R 4.1.1 so this is recommended. As for plotting, there are some issues with the WEXAC server so save the files and run the plots and pdf creations from your computer (unless you know how to save it)]{style="font-size: 10pt"}

```{r, eval = FALSE}
if(!require("devtools")){install.packages("devtools")}

devtools::install_github("TomerAntman/VCFtoGWAS") 
```
</div>



<!-- ### Output structure -->
<div id="Output structure" class="tabcontent">
<h3>Output structure</h3>

```{r, echo = FALSE, message=FALSE, warning=FALSE}
show_suggested_data_structure()
```
</div>

<!-- ### VCF info -->
<div id="VCF info" class="tabcontent">
<h3>VCF info</h3>
VCF (Variant Call Format) is a text file format (most likely stored in a compressed manner). It contains meta-information lines, a header line, and then data lines each containing information about a position in the genome. The format also has the ability to contain genotype information on samples for each position. For more info, I suggest reading the [VCF File-format specifications](https://samtools.github.io/hts-specs/VCFv4.2.pdf) by samtools or [Reading VCF data](https://grunwaldlab.github.io/Population_Genetics_in_R/reading_vcf.html) by BJ Knaus, JF Tabima and NJ Grünwald.

The cartoon below also demonstrates the division of a VCF file into 3 sections:

1. Meta information (@meta): containing information about the file, its creation, as well as information needed to interpert abbreviations used elsewhere in the file.
2. Fixed information (@fix): containing information for each variant (SNP or INDEL).
3. Genotype information (@gt): containing information about each variant for each sample (such as "strain" or "individual")

<center>

![VCF Cartoon](https://grunwaldlab.github.io/Population_Genetics_in_R/reading_vcf_files/figure-html/unnamed-chunk-1-1.png)
</center>

A demonstration of how the results appear in the code (as the VCF is loaded using the vcfR package)

![VCF Cartoon](https://github.com/TomerAntman/VCFtoGWAS/tree/main/man/figures/VCF_demo1.png)
</center>


</div>
</body>
