---
title: "General"
author: "Tomer Antman"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{General}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r setup, echo=FALSE}
library(VCFtoGWAS)
```

## Motivation

#### *GWAS*

The aim of Genome‐wide association studies (GWAS) is to identify single nucleotide polymorphisms (SNPs) of which the allele frequencies vary systematically as a function of phenotypic trait values.

Identification of trait‐associated SNPs may subsequently reveal new insights into the biological mechanisms underlying these phenotypes.

The main package used here for GWAS is "[statgenGWAS](https://cran.r-project.org/web/packages/statgenGWAS/vignettes/GWAS.html)"

#### *VCF*

VCF files are big genetic files and working with them might be complicated.

The main package used here for loading the vcf file is "[vcfR](https://cran.r-project.org/web/packages/vcfR/vignettes/)"

### Usage

If a server with stronger computation is available, it is required to run most of the functions on it (especially chapter 1).\
For now, the R markdown files that show how to use the package step by step (chapter 1 and chapter 2) aren't publicly available. You can send an email [the maintainer]() of the package

[For Weizmann institute users: The package is recommended to be executed on the WEXAC R server. All the required packages are installed on R 4.1.1 so this is recommended. As for plotting, there are some issues with the WEXAC server so save the files and run the plots and pdf creations from your computer (unless you know how to save it)]{style="font-size: 10pt"}

### Installing the package

```{r, eval = FALSE}
if(!require("devtools")){install.packages("devtools")}

devtools::install_github("TomerAntman/VCFtoGWAS") 
```

### Output structure

```{r, echo = FALSE, message=FALSE, warning=FALSE}
show_suggested_data_structure()
```
