<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="VCFtoGWAS">
<title>General • VCFtoGWAS</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="General">
<meta property="og:description" content="VCFtoGWAS">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">VCFtoGWAS</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.2.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Functions</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Manual</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/1)%20General.html">1) General</a>
    <a class="dropdown-item" href="../articles/2)%20Chapter%201.html">2) Chapter 1</a>
    <a class="dropdown-item" href="../articles/3)%20Chapter%202.html">3) Chapter 2</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>General</h1>
            
      
      
      <div class="d-none name"><code>1) General.Rmd</code></div>
    </div>

    
    
<!-- Website built with pkgdown::build_site()  -->
<div class="section level2 tabset">
<h2 class="tabset" id="motivation">Motivation<a class="anchor" aria-label="anchor" href="#motivation"></a>
</h2>


<ul class="nav nav-tabs" id="motivation" role="tablist">
<li role="presentation" class="nav-item"><button data-bs-toggle="tab" data-bs-target="#gwas" id="gwas-tab" type="button" role="tab" aria-controls="gwas" aria-selected="true" class="active nav-link">GWAS</button></li>
<li role="presentation" class="nav-item"><button data-bs-toggle="tab" data-bs-target="#vcf" id="vcf-tab" type="button" role="tab" aria-controls="vcf" aria-selected="false" class="nav-link">VCF</button></li>
</ul>
<div class="tab-content">
<div class="active  tab-pane" id="gwas" role="tabpanel" aria-labelledby="gwas-tab">

<p>The aim of Genome‐wide association studies (GWAS) is to identify single nucleotide polymorphisms (SNPs) of which the allele frequencies vary systematically as a function of phenotypic trait values.</p>
<p>Identification of trait‐associated SNPs may subsequently reveal new insights into the biological mechanisms underlying these phenotypes.</p>
</div>
<div class="tab-pane" id="vcf" role="tabpanel" aria-labelledby="vcf-tab">

<p>VCF files are big, commonly used, genetic files and working with them might be complicated. More information on VCF format below.</p>
<p>The main package used here for loading the vcf file is “<a href="https://cran.r-project.org/web/packages/vcfR/vignettes/" class="external-link">vcfR</a>”</p>
<hr>
</div>
</div>
</div>
<div class="section level2">
<h2 id="usage">Usage<a class="anchor" aria-label="anchor" href="#usage"></a>
</h2>
<p>If a server with stronger computation is available, it is required to run most of the functions on it (especially chapter 1).<br>
For now, the R markdown files that show how to use the package step by step (chapter 1 and chapter 2) aren’t publicly available. You can send an email <a href="">the maintainer</a> of the package</p>
<p><span style="font-size: 10pt">For Weizmann institute users: The package is recommended to be executed on the WEXAC R server. All the required packages are installed on R 4.1.1 so this is recommended. As for plotting, there are some issues with the WEXAC server so save the files and run the plots and pdf creations from your computer (unless you know how to save it)</span></p>
<hr>
</div>
<div class="section level2">
<h2 id="installing-the-package">Installing the package<a class="anchor" aria-label="anchor" href="#installing-the-package"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">if</span><span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://devtools.r-lib.org/" class="external-link">"devtools"</a></span><span class="op">)</span><span class="op">)</span><span class="op">{</span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span><span class="op">}</span>

<span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org/reference/remote-reexports.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"TomerAntman/VCFtoGWAS"</span><span class="op">)</span> </code></pre></div>
<hr>
</div>
<div class="section level2">
<h2 id="output-structure">Output structure<a class="anchor" aria-label="anchor" href="#output-structure"></a>
</h2>
<pre><code><span class="co">##                                           levelName</span>
<span class="co">## 1  Results                                         </span>
<span class="co">## 2   ¦--Chapter1-VCF2GWAS                           </span>
<span class="co">## 3   ¦   °--Step1.1-Upload_VCF                      </span>
<span class="co">## 4   ¦       ¦--fix_sub.RDS                         </span>
<span class="co">## 5   ¦       ¦--gt_GTonly.RDS                       </span>
<span class="co">## 6   ¦       °--Step1.2-Filter_genotypes            </span>
<span class="co">## 7   ¦           ¦--fix_filt.RDS                    </span>
<span class="co">## 8   ¦           ¦--gt_GTonly_filt.RDS              </span>
<span class="co">## 9   ¦           °--Step1.3-Expand                  </span>
<span class="co">## 10  ¦               ¦--fix_filt_expand.RDS         </span>
<span class="co">## 11  ¦               ¦--indication.RDS              </span>
<span class="co">## 12  ¦               ¦--gt_GTonly_filt_expand.RDS   </span>
<span class="co">## 13  ¦               °--Step1.4-GWAS_Matrix         </span>
<span class="co">## 14  ¦                   ¦--mapping_info.RDS        </span>
<span class="co">## 15  ¦                   ¦--GWAS_mat.RDS            </span>
<span class="co">## 16  ¦                   ¦--phenotypes.csv          </span>
<span class="co">## 17  ¦                   ¦--offspring_GWAS_mat.RDS  </span>
<span class="co">## 18  ¦                   °--offspring_phenotypes.csv</span>
<span class="co">## 19  °--Chapter2-Analysis                           </span>
<span class="co">## 20      °--Step2.1-Create_gData                    </span>
<span class="co">## 21          ¦--gData.RDS                           </span>
<span class="co">## 22          °--Step2.2-Single_Trait_GWAS           </span>
<span class="co">## 23              ¦--GWAS_Result.RDS                 </span>
<span class="co">## 24              ¦--Significant_SNPs_information.csv</span>
<span class="co">## 25              °--complete_results_plots.pdf</span></code></pre>
<hr>
</div>
<div class="section level2">
<h2 id="vcf-info">VCF info<a class="anchor" aria-label="anchor" href="#vcf-info"></a>
</h2>
<p>VCF (Variant Call Format) is a text file format (most likely stored in a compressed manner).
    It contains meta-information lines, a header line, and then data lines each containing information about a position in the genome.
    The format also has the ability to contain genotype information on samples for each position. For more info, I suggest reading the
    <a href="https://samtools.github.io/hts-specs/VCFv4.2.pdf" class="external-link">VCF File-format specifications</a> by samtools or
    <a href="https://grunwaldlab.github.io/Population_Genetics_in_R/reading_vcf.html" class="external-link">Reading VCF data</a> by BJ Knaus, JF Tabima and NJ Grünwald.</p>
<p>The cartoon below also demonstrates the division of a VCF file into 3 sections:</p>
<p>1. Meta information (@meta): containing information about the file, its creation, as well as information needed to interpert abbreviations used elsewhere in the file.</p>
<p>2. Fixed information (@fix): containing information for each variant (SNP or INDEL).</p>
<p>3. Genotype information (@gt): containing information about each variant for each sample (such as “strain” or “individual”)</p>
<!-- Control the size of plots/images: https://bookdown.org/yihui/rmarkdown-cookbook/figure-size.html-->
<center>
<div class="figure">
<img src="https://grunwaldlab.github.io/Population_Genetics_in_R/reading_vcf_files/figure-html/unnamed-chunk-1-1.png" size="50%" alt=""><p class="caption">VCF Cartoon</p>
</div>
</center>
<p>A demonstration of how the results appear in the code (as the VCF is loaded using the vcfR package):</p>
<center>
<p><img src="./VCF_demo1.png"></p>
</center>
<hr>
</div>
<div class="section level2">
<h2 id="r-cheats-and-tips">R cheats and tips<a class="anchor" aria-label="anchor" href="#r-cheats-and-tips"></a>
</h2>
<div class="section level4">
<h4 id="rds">RDS<a class="anchor" aria-label="anchor" href="#rds"></a>
</h4>
<p>The easiest way to save file for further use in R is by saving them as “.RDS” files (I saw there are python packages for reading RDS files but haven’t tried).</p>
<p>All you need to do to save is run:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">saveRDS</a></span><span class="op">(</span><span class="va">some_object</span>, <span class="st">"some_path/some_object.RDS"</span><span class="op">)</span></code></pre></div>
<p>All you need to do to load is run:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">some_object</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">"some_path/some_object.RDS"</span><span class="op">)</span></code></pre></div>
<p>Sometimes when reading an RDS file it is uploaded as a list containing the desired object.</p>
<p>If that happens, run:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">some_object</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">"some_path/some_object.RDS"</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></code></pre></div>
</div>
<div class="section level4">
<h4 id="paths">Paths<a class="anchor" aria-label="anchor" href="#paths"></a>
</h4>
<p>There is an issue with directories in R. It accepts only forward slashes in paths (/).</p>
<p>A cool cheat around this is to copy the full file path (you can use Shift + Right click on a file which opens a menu with the “Copy as path” option) and then run the following line (in some constellation):</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">'"'</span>,<span class="st">""</span>,<span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"\\\\\\\\"</span>,<span class="st">"/"</span>,<span class="fu"><a href="https://rdrr.io/r/utils/clipboard.html" class="external-link">readClipboard</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p>For example, what I did is put it into a function in the beginning of markdown like this:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">paste_directory</span><span class="op">&lt;-</span><span class="kw">function</span><span class="op">(</span><span class="op">)</span><span class="op">{</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">'"'</span>,<span class="st">""</span>,<span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"\\\\\\\\"</span>,<span class="st">"/"</span>,<span class="fu"><a href="https://rdrr.io/r/utils/clipboard.html" class="external-link">readClipboard</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">}</span></code></pre></div>
<p>And then every time I copy the directory and just write in the Rstudio console: <code>paste_directory()</code></p>
</div>
<div class="section level4">
<h4 id="common-error">Common error<a class="anchor" aria-label="anchor" href="#common-error"></a>
</h4>
<p style="color:red">Error: cannot allocate vector of size ... </p>
<p>The solution is increasing the memory limit by running <code><a href="https://rdrr.io/r/utils/memory.size.html" class="external-link">memory.limit()</a></code> to check the current limitation and then increasing it by <code>memory.limit(size = ...)</code></p>
<hr>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Tomer Antman.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.9000.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
