<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="VCFtoGWAS">
<title>package_intro • VCFtoGWAS</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="package_intro">
<meta property="og:description" content="VCFtoGWAS">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">VCFtoGWAS</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.2</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/General.html">General</a>
    <a class="dropdown-item" href="../articles/package_intro.html">package_intro</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main"><div class="page-header">
      <img src="" class="logo" alt=""><h1>package_intro</h1>
                        <h4 data-toc-skip class="author">Tomer Antman</h4>
            
      
      
      <div class="d-none name"><code>package_intro.Rmd</code></div>
    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">VCFtoGWAS</span><span class="op">)</span></code></pre></div>
<div class="section level2">
<h2 class="tabset" id="package-manual">Package Manual<a class="anchor" aria-label="anchor" href="#package-manual"></a>
</h2>
<div class="section level3">
<h3 class="tabset" id="general">General<a class="anchor" aria-label="anchor" href="#general"></a>
</h3>
<div class="section level4">
<h4 id="motivation">Motivation<a class="anchor" aria-label="anchor" href="#motivation"></a>
</h4>
<div class="section level5">
<h5 id="gwas">
<em>GWAS</em><a class="anchor" aria-label="anchor" href="#gwas"></a>
</h5>
<p>The aim of Genome‐wide association studies (GWAS) is to identify single nucleotide polymorphisms (SNPs) of which the allele frequencies vary systematically as a function of phenotypic trait values.</p>
<p>Identification of trait‐associated SNPs may subsequently reveal new insights into the biological mechanisms underlying these phenotypes.</p>
<p>The main package used here for GWAS is “<a href="https://cran.r-project.org/web/packages/statgenGWAS/vignettes/GWAS.html" class="external-link">statgenGWAS</a>”</p>
</div>
<div class="section level5">
<h5 id="vcf">
<em>VCF</em><a class="anchor" aria-label="anchor" href="#vcf"></a>
</h5>
<p>VCF files are big genetic files and working with them might be complicated.</p>
<p>The main package used here for loading the vcf file is “<a href="https://cran.r-project.org/web/packages/vcfR/vignettes/" class="external-link">vcfR</a>”</p>
</div>
</div>
<div class="section level4">
<h4 id="usage">Usage<a class="anchor" aria-label="anchor" href="#usage"></a>
</h4>
<p>If a server with stronger computation is available, it is required to run most of the functions on it (especially chapter 1).<br>
For now, the R markdown files that show how to use the package step by step (chapter 1 and chapter 2) aren’t publicly available. You can send an email <a href="">the maintainer</a> of the package</p>
<p>|For Weizmann institute users: |The package is recommended to be executed on the WEXAC R server. |All the required packages are installed on R 4.1.1 so this is recommended. |As for plotting, there are some issues with the WEXAC server so save the files and run the plots and pdf creations from your computer (unless you know how to save it)</p>
</div>
<div class="section level4">
<h4 id="installing-the-package">Installing the package<a class="anchor" aria-label="anchor" href="#installing-the-package"></a>
</h4>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">require</span>(<span class="st">"devtools"</span>)){<span class="fu">install.packages</span>(<span class="st">"devtools"</span>)}</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Loading required package: devtools</span></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Loading required package: usethis</span></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">install_github</span>(<span class="st">"TomerAntman/VCFtoGWAS"</span>) </span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Downloading GitHub repo TomerAntman/VCFtoGWAS@HEAD</span></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; dplyr (1.0.7 -&gt; 1.0.8) [CRAN]</span></span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Installing 1 packages: dplyr</span></span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Installing package into 'C:/Users/tomer/AppData/Local/Temp/Rtmp4uji5m/temp_libpath4104538814a2'</span></span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; (as 'lib' is unspecified)</span></span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   There is a binary version available but the source version is later:</span></span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;       binary source needs_compilation</span></span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; dplyr  1.0.7  1.0.8              TRUE</span></span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; installing the source package 'dplyr'</span></span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   </span></span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb2-19"><a href="#cb2-19" aria-hidden="true" tabindex="-1"></a>   checking <span class="cf">for</span> file <span class="st">'C:\Users</span><span class="sc">\t</span><span class="st">omer\AppData\Local\Temp\RtmpmytCsH</span><span class="sc">\r</span><span class="st">emotes14c861d74d7e\TomerAntman-VCFtoGWAS-e736d0d/DESCRIPTION'</span> ...</span>
<span id="cb2-20"><a href="#cb2-20" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb2-21"><a href="#cb2-21" aria-hidden="true" tabindex="-1"></a> [32mv [39m   [<span class="dv">38</span>;<span class="dv">5</span>;247mchecking <span class="cf">for</span> file <span class="st">'C:\Users</span><span class="sc">\t</span><span class="st">omer\AppData\Local\Temp\RtmpmytCsH</span><span class="sc">\r</span><span class="st">emotes14c861d74d7e\TomerAntman-VCFtoGWAS-e736d0d/DESCRIPTION'</span> [39m [36m [39m</span>
<span id="cb2-22"><a href="#cb2-22" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-23"><a href="#cb2-23" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb2-24"><a href="#cb2-24" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb2-25"><a href="#cb2-25" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb2-26"><a href="#cb2-26" aria-hidden="true" tabindex="-1"></a> [<span class="dv">38</span>;<span class="dv">5</span>;247m<span class="sc">-</span> [39m [<span class="dv">38</span>;<span class="dv">5</span>;247m   [39m [<span class="dv">38</span>;<span class="dv">5</span>;247mpreparing <span class="st">'VCFtoGWAS'</span><span class="sc">:</span> [39m [36m [36m (481ms) [36m [39m</span>
<span id="cb2-27"><a href="#cb2-27" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;    checking DESCRIPTION meta-information ...</span></span>
<span id="cb2-28"><a href="#cb2-28" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb2-29"><a href="#cb2-29" aria-hidden="true" tabindex="-1"></a>   checking DESCRIPTION meta<span class="sc">-</span>information ... </span>
<span id="cb2-30"><a href="#cb2-30" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb2-31"><a href="#cb2-31" aria-hidden="true" tabindex="-1"></a> [32mv [39m   [<span class="dv">38</span>;<span class="dv">5</span>;247mchecking DESCRIPTION meta<span class="sc">-</span>information [39m [36m [39m</span>
<span id="cb2-32"><a href="#cb2-32" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-33"><a href="#cb2-33" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb2-34"><a href="#cb2-34" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb2-35"><a href="#cb2-35" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb2-36"><a href="#cb2-36" aria-hidden="true" tabindex="-1"></a> [<span class="dv">38</span>;<span class="dv">5</span>;247m<span class="sc">-</span> [39m [<span class="dv">38</span>;<span class="dv">5</span>;247m   [39m [<span class="dv">38</span>;<span class="dv">5</span>;247mchecking <span class="cf">for</span> LF line<span class="sc">-</span>endings <span class="cf">in</span> source and make files and shell scripts [39m [36m [39m</span>
<span id="cb2-37"><a href="#cb2-37" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-38"><a href="#cb2-38" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb2-39"><a href="#cb2-39" aria-hidden="true" tabindex="-1"></a> [<span class="dv">38</span>;<span class="dv">5</span>;247m<span class="sc">-</span> [39m [<span class="dv">38</span>;<span class="dv">5</span>;247m   [39m [<span class="dv">38</span>;<span class="dv">5</span>;247mchecking <span class="cf">for</span> empty or unneeded directories [39m [36m [39m</span>
<span id="cb2-40"><a href="#cb2-40" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-41"><a href="#cb2-41" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb2-42"><a href="#cb2-42" aria-hidden="true" tabindex="-1"></a>   NB<span class="sc">:</span> this package now depends on <span class="fu">R</span> (<span class="sc">&gt;=</span> </span>
<span id="cb2-43"><a href="#cb2-43" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb2-44"><a href="#cb2-44" aria-hidden="true" tabindex="-1"></a>     NB<span class="sc">:</span> this package now depends on <span class="fu">R</span> (<span class="sc">&gt;=</span> <span class="dv">3</span>.<span class="fl">5.0</span>)</span>
<span id="cb2-45"><a href="#cb2-45" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;      </span><span class="al">WARNING</span><span class="co">: Added dependency on R &gt;= 3.5.0 because serialized objects in</span></span>
<span id="cb2-46"><a href="#cb2-46" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;      serialize/load version 3 cannot be read in older versions of R.</span></span>
<span id="cb2-47"><a href="#cb2-47" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;      File(s) containing such objects:</span></span>
<span id="cb2-48"><a href="#cb2-48" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;        'VCFtoGWAS/data/SacCer_sgd_cds.rda'</span></span>
<span id="cb2-49"><a href="#cb2-49" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-50"><a href="#cb2-50" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb2-51"><a href="#cb2-51" aria-hidden="true" tabindex="-1"></a> [<span class="dv">38</span>;<span class="dv">5</span>;247m<span class="sc">-</span> [39m [<span class="dv">38</span>;<span class="dv">5</span>;247m   [39m [<span class="dv">38</span>;<span class="dv">5</span>;247mbuilding <span class="st">'VCFtoGWAS_0.1.2.tar.gz'</span> [39m [36m [39m</span>
<span id="cb2-52"><a href="#cb2-52" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-53"><a href="#cb2-53" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb2-54"><a href="#cb2-54" aria-hidden="true" tabindex="-1"></a>   </span>
<span id="cb2-55"><a href="#cb2-55" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-56"><a href="#cb2-56" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: package 'VCFtoGWAS' is in use and will not be installed</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="output-structure">Output structure<a class="anchor" aria-label="anchor" href="#output-structure"></a>
</h4>
<pre><code><span class="co">#&gt; Loading required package: data.tree</span>
<span class="co">#&gt;                                           levelName</span>
<span class="co">#&gt; 1  Results                                         </span>
<span class="co">#&gt; 2   ¦--Chapter1-VCF2GWAS                           </span>
<span class="co">#&gt; 3   ¦   °--Step1.1-Upload_VCF                      </span>
<span class="co">#&gt; 4   ¦       ¦--fix_sub.RDS                         </span>
<span class="co">#&gt; 5   ¦       ¦--gt_GTonly.RDS                       </span>
<span class="co">#&gt; 6   ¦       °--Step1.2-Filter_genotypes            </span>
<span class="co">#&gt; 7   ¦           ¦--fix_filt.RDS                    </span>
<span class="co">#&gt; 8   ¦           ¦--gt_GTonly_filt.RDS              </span>
<span class="co">#&gt; 9   ¦           °--Step1.3-Expand                  </span>
<span class="co">#&gt; 10  ¦               ¦--fix_filt_expand.RDS         </span>
<span class="co">#&gt; 11  ¦               ¦--indication.RDS              </span>
<span class="co">#&gt; 12  ¦               ¦--gt_GTonly_filt_expand.RDS   </span>
<span class="co">#&gt; 13  ¦               °--Step1.4-GWAS_Matrix         </span>
<span class="co">#&gt; 14  ¦                   ¦--mapping_info.RDS        </span>
<span class="co">#&gt; 15  ¦                   ¦--GWAS_mat.RDS            </span>
<span class="co">#&gt; 16  ¦                   ¦--phenotypes.csv          </span>
<span class="co">#&gt; 17  ¦                   ¦--offspring_GWAS_mat.RDS  </span>
<span class="co">#&gt; 18  ¦                   °--offspring_phenotypes.csv</span>
<span class="co">#&gt; 19  °--Chapter2-Analysis                           </span>
<span class="co">#&gt; 20      °--Step2.1-Create_gData                    </span>
<span class="co">#&gt; 21          ¦--gData.RDS                           </span>
<span class="co">#&gt; 22          °--Step2.2-Single_Trait_GWAS           </span>
<span class="co">#&gt; 23              ¦--GWAS_Result.RDS                 </span>
<span class="co">#&gt; 24              ¦--Significant_SNPs_information.csv</span>
<span class="co">#&gt; 25              °--complete_results_plots.pdf</span></code></pre>
</div>
</div>
<div class="section level3">
<h3 class="tabset" id="chapter-1---vcf-file-to-gwas-matrix">Chapter 1 - VCF file to GWAS matrix<a class="anchor" aria-label="anchor" href="#chapter-1---vcf-file-to-gwas-matrix"></a>
</h3>
<div class="section level4">
<h4 id="step-1-1---upload-vcf-to-r">Step 1.1 - Upload VCF to R<a class="anchor" aria-label="anchor" href="#step-1-1---upload-vcf-to-r"></a>
</h4>
<p>…</p>
</div>
<div class="section level4">
<h4 id="step-1-2---filter">Step 1.2 - Filter<a class="anchor" aria-label="anchor" href="#step-1-2---filter"></a>
</h4>
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Tomer Antman.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.9000.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
