<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content='This function serves for step 1 of chapter 1.1 of the "Using VCFtoGWAS package" markdown series.
Based on the GWAS matrix of the parents (assuming the matrix was built based on diploid parents), a matrix is built for the offspring of the parents.
'><title>Build a GWAS matrix for the offspring
 — Build_offspring_GWAS_matrix • VCFtoGWAS</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Build a GWAS matrix for the offspring
 — Build_offspring_GWAS_matrix"><meta property="og:description" content='This function serves for step 1 of chapter 1.1 of the "Using VCFtoGWAS package" markdown series.
Based on the GWAS matrix of the parents (assuming the matrix was built based on diploid parents), a matrix is built for the offspring of the parents.
'><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">VCFtoGWAS</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.2</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/1)%20General.html">General</a>
    <a class="dropdown-item" href="../articles/2)%20Chapter%201.html">Chapter 1 - VCF file to GWAS matrix</a>
    <a class="dropdown-item" href="../articles/3)%20Chapter%202.html">Chapter 2 - Run GWAS and Analyze</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"></ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Build a GWAS matrix for the offspring
<!-- %%  ~~function to do ... ~~ --></h1>
      
      <div class="d-none name"><code>Build_offspring_GWAS_matrix.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This function serves for step 1 of chapter 1.1 of the "Using VCFtoGWAS package" markdown series.<br>
Based on the GWAS matrix of the parents (assuming the matrix was built based on diploid parents), a matrix is built for the offspring of the parents.
<!-- %%  ~~ A concise (1-5 lines) description of what the function does. ~~ --></p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">Build_offspring_GWAS_matrix</span><span class="op">(</span><span class="va">parent_GWAS_mat</span>,
                            <span class="va">offspring_strains</span>,
                            dir_results <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/getwd.html" class="external-link">getwd</a></span><span class="op">(</span><span class="op">)</span>,
                            results_name <span class="op">=</span> <span class="fu"><a href="name_by_time.html">name_by_time</a></span><span class="op">(</span><span class="op">)</span>,
                            do_save <span class="op">=</span> <span class="cn">TRUE</span>,
                            seed <span class="op">=</span> <span class="fl">10</span>,
                            save_hashed_matrix <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>parent_GWAS_mat</dt>
<dd><p>A GWAS matrix (composed of 0,1,2,NA) where the rownames are the parent strains</p></dd>
<dt>offspring_strains</dt>
<dd><p>An array of the names of the offspring, must be composed of the parent names</p></dd>
<dt>dir_results</dt>
<dd><p>The directory in which a folder will be created and results will be saved. Make sure it exists!!!</p></dd>
<dt>results_name</dt>
<dd><p>The name of the folder in which the results will be saved within dir_results (default is a time stamp, see <code><a href="create_directory.html">create_directory</a></code> function)</p></dd>
<dt>do_save</dt>
<dd><p>Do you wish to save the results? (will be saved as RDS files) (Default is TRUE)</p></dd>
<dt>seed</dt>
<dd><p>The function includes random sampling. A seed can be set and changed.</p></dd>
<dt>save_hashed_matrix</dt>
<dd><p>Do you wish to also save the the "mid-product" of the function (the hashed matrix). Default is <code>FALSE</code></p></dd>
</dl><p><!-- %%     ~~Describe \code{x} here~~ --></p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>Hashing the table:<br>
If parent1 is "BNI" and parent2 is "BKE" than the offspring name is "BNIBKE".<br>
A hashed value for "BNIBKE" (denoted BNIBKE^[i,j]) is given by:<br>
BNIBKE^[i,j] = (BNI[i,j] + BKE[i,j]) + (BNI[i,j] * BKE[i,j])<br></p>
<p>Un-hashing the table:<br></p><ul><li><p>(0 + 0) + (0 * 0) = 0 ==&gt; BNIBKE[i,j] = 0</p></li>
<li><p>(0 + 1) + (0 * 1) = 1 ==&gt; BNIBKE[i,j] = 0 or 1</p></li>
<li><p>(0 + 2) + (0 * 2) = 2 ==&gt; BNIBKE[i,j] = 1</p></li>
<li><p>(1 + 1) + (1 * 1) = 3 ==&gt; BNIBKE[i,j] = 0 or 1 or 2</p></li>
<li><p>(1 + 2) + (1 * 2) = 5 ==&gt; BNIBKE[i,j] = 1 or 2</p></li>
<li><p>(2 + 2) + (2 * 2) = 8 ==&gt; BNIBKE[i,j] = 2</p></li>
<li><p>NA in any of them results in NA</p></li>
</ul></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>The returned object is a list of length 3 or 1:</p>
<dl><dt>offspring_GWAS_mat</dt>
<dd><p>The created GWAS matrix of the offspring (always returned)</p></dd>
<dt>mapping_info_offspring</dt>
<dd><p>The mapping info linking between the SNPs and their location (after the creation of the new matrix, some SNPs are removed)</p></dd>
<dt>results_directory</dt>
<dd><p>If the results are saved, this is the directory where they were saved</p></dd>
</dl></div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>Tomer Antman</p>
    </div>

  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Tomer Antman.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.9000.</p>
</div>

    </footer></div>

  

  

  </body></html>

