<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content='This function serves for step 3 of chapter 1 of the "Using VCFtoGWAS package" markdown series.
Expand the fixed info and the genotype (gt) matrix so that each row represents only one SNP or INDEL (up until now each row represented a position)
The output is the expanded fixed and gt and also an array (called `indication` that links between the two. Relevant for the next step)'><title>Expand rows of files — Expand_files • VCFtoGWAS</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Expand rows of files — Expand_files"><meta property="og:description" content='This function serves for step 3 of chapter 1 of the "Using VCFtoGWAS package" markdown series.
Expand the fixed info and the genotype (gt) matrix so that each row represents only one SNP or INDEL (up until now each row represented a position)
The output is the expanded fixed and gt and also an array (called `indication` that links between the two. Relevant for the next step)'><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">VCFtoGWAS</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.2</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/General.html">General</a>
    <a class="dropdown-item" href="../articles/package_intro.html">package_intro</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"></ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Expand rows of files</h1>
      
      <div class="d-none name"><code>Expand_files.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This function serves for step 3 of chapter 1 of the "Using VCFtoGWAS package" markdown series.<br>
Expand the fixed info and the genotype (gt) matrix so that each row represents only one SNP or INDEL (up until now each row represented a position)
The output is the expanded fixed and gt and also an array (called `indication` that links between the two. Relevant for the next step)</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">expand_files</span><span class="op">(</span><span class="va">gt_GTonly_filt</span>,
             <span class="va">fix_filt</span>,
             dir_results <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/getwd.html" class="external-link">getwd</a></span><span class="op">(</span><span class="op">)</span>,
             results_name <span class="op">=</span> <span class="fu"><a href="name_by_time.html">name_by_time</a></span><span class="op">(</span><span class="op">)</span>,
             do_save <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>gt_GTonly_filt</dt>
<dd><p>The filtered Genotype matrix (step2 output)</p></dd>
<dt>fix_filt</dt>
<dd><p>The filtered fixed information dataframe (step2 output)</p></dd>
<dt>dir_results</dt>
<dd><p>The directory in which a folder will be created and results will be saved. Make sure it exists!!!</p></dd>
<dt>results_name</dt>
<dd><p>The name of the folder in which the results will be saved within dir_results (default is a time stamp, see <code><a href="create_directory.html">create_directory</a></code>)</p></dd>
<dt>do_save</dt>
<dd><p>Do you wish to save the results? (will be saved as RDS files) (Default is TRUE)</p></dd>
</dl></div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>For very large files it might take a while.<br><br>
An explanation about the <code>indication</code> array that is returned:<br>
If for example, row number 3 in <code>fix_filt</code> represents some position that has two ALTs (like: [C, TG]) for a REF (like: [A]),  in <code>fix_filt_expand</code> it will be two rows (as 3.0, 3.1) with only one ALT in each.<br>
The <code>indication</code> array will get a "1" for row 3.0 and "2" for row 3.1. This array is the link between the data in the <code>fix_filt_expand</code> and the data in <code>gt_GTonly_filt_expand</code>.<br>
The <code>indication[i]</code> gives us the id number of the alternative sequence in a certain location.<br>
It is between 1 and some number (specifically here it's 6 but can be more).<br>
In simpler terms: after all the pre-processing, each row in <code>fix_filt_expand</code> represents only one alteration from the reference (only one ALT). Since the same position might have had several alterations (in <code>fix_filt</code>), the <code>indication</code> array indicates what alteration is represented in each row (in regards to the position in the genome).</p>
    </div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <dl><dt>fix_filt_expand</dt>
<dd><p>A dataframe of the fixed information. Each row represents only one variant(one SNP or INDEL). It has an extra column: `ALT_options` which represents the number of alterations that exist per genomic position in the data</p></dd>
<dt>gt_GTonly_filt_expand</dt>
<dd><p>A matrix in which the number of times each row appears is affected by the `ALT_options` in <code>fix_filt_expand</code></p></dd>
<dt>indication</dt>
<dd><p>This array is the link between the data in the <code>fix_filt_expand</code> and the data in <code>gt_GTonly_filt_expand</code></p></dd>
</dl></div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>Tomer Antman</p>
    </div>
    <div class="section level2">
    <h2 id="note">Note<a class="anchor" aria-label="anchor" href="#note"></a></h2>
    <p>This should be run on the results of the <code><a href="Filter_genotypes.html">Filter_genotypes</a></code> function from step 2.</p>
    </div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p><!-- %% ~~objects to See Also as \code{\link{help}}, ~~~ --></p></div>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"></span>
<span class="r-in"></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Tomer Antman.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.9000.</p>
</div>

    </footer></div>

  

  

  </body></html>

