<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content='This function serves for step 1 of chapter 1 of the "Using VCFtoGWAS package" markdown series.
This script loads the vcf file that you want to work with  (either processed by the "usegalaxy" server or not). The file can be uploaded in its zipped version (.gz file extension). The vcf processing is based on the [vcfR package](https://cran.r-project.org/web/packages/vcfR/vignettes/intro_to_vcfR.html).'><title>Turn vcf files to usable R files
 — Upload_vcf_to_R • VCFtoGWAS</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Turn vcf files to usable R files
 — Upload_vcf_to_R"><meta property="og:description" content='This function serves for step 1 of chapter 1 of the "Using VCFtoGWAS package" markdown series.
This script loads the vcf file that you want to work with  (either processed by the "usegalaxy" server or not). The file can be uploaded in its zipped version (.gz file extension). The vcf processing is based on the [vcfR package](https://cran.r-project.org/web/packages/vcfR/vignettes/intro_to_vcfR.html).'><meta property="og:image" content="/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">VCFtoGWAS</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.2.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/1)%20General.html">General</a>
    <a class="dropdown-item" href="../articles/2)%20Chapter%201.html">Chapter 1 - VCF file to GWAS matrix</a>
    <a class="dropdown-item" href="../articles/3)%20Chapter%202.html">Chapter 2 - Run GWAS and Analyze</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"></ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Turn vcf files to usable R files
<!-- %%  ~~function to do ... ~~ --></h1>
      
      <div class="d-none name"><code>Upload_vcf_to_R.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This function serves for step 1 of chapter 1 of the "Using VCFtoGWAS package" markdown series.<br>
This script loads the vcf file that you want to work with  (either processed by the "usegalaxy" server or not). The file can be uploaded in its zipped version (.gz file extension). The vcf processing is based on the [vcfR package](https://cran.r-project.org/web/packages/vcfR/vignettes/intro_to_vcfR.html).</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">Upload_vcf_to_R</span><span class="op">(</span><span class="va">vcf_file</span>,
                dir_results <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/getwd.html" class="external-link">getwd</a></span><span class="op">(</span><span class="op">)</span>,
                results_name <span class="op">=</span> <span class="fu"><a href="name_by_time.html">name_by_time</a></span><span class="op">(</span><span class="op">)</span>,
                do_save <span class="op">=</span> <span class="cn">TRUE</span>,
                do_return <span class="op">=</span> <span class="cn">TRUE</span>,
                get_chr_info <span class="op">=</span> <span class="cn">TRUE</span>,
                fix_columns <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"CHROM"</span>,<span class="st">"POS"</span>,<span class="st">"REF"</span>,<span class="st">"ALT"</span>,<span class="st">"QUAL"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>vcf_file</dt>
<dd><p>The location where the vcf file is saved</p></dd>
<dt>dir_results</dt>
<dd><p>The directory in which a subfolder will be created and results will be saved. Make sure it exists!!!</p></dd>
<dt>results_name</dt>
<dd><p>The name of the folder in which the results will be saved within dir_results (default is a time stamp, see <code><a href="create_directory.html">create_directory</a></code>)</p></dd>
<dt>do_save</dt>
<dd><p>Do you wish to save the results? (will be saved as RDS files) (Default is TRUE)</p></dd>
<dt>do_return</dt>
<dd><p>Do you wish to return the results to your current workspace environment? (Default is TRUE)</p></dd>
<dt>get_chr_info</dt>
<dd><p>Information regarding the lengths of the chromosomes (Default is TRUE)</p></dd>
<dt>fix_columns</dt>
<dd><p>The column names that you wish to get from the fixed column of the vcf (the default is probably all you need so don't change it)</p></dd>
</dl></div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>These files are usually very large and it will take a while.<br>
The files exported are saved as .RDS files. They are lighter and very easy to read in R by calling readRDS(file = filepath).<br><br>
Extract genotypes from the vcf data:<br>
GT: genotype, encoded as allele values separated by either of / or |.<br>
The allele values are:</p><ul><li><p>0 for the reference allele (what is in the REF field)</p></li>
<li><p>1 for the first allele listed in ALT</p></li>
<li><p>2 for the second allele listed in ALT.</p></li>
<li><p>3 for the third allele listed in ALT and so on.</p></li>
</ul><p>For diploid calls examples could be 0/1, 1|0, or 1/2, etc. If a call cannot be made for a sample at a given locus, '.' is specified for each missing allele in the GT field (for example './.' for a diploid genotype and '.' for haploid genotype).<br>
The meanings of the separators are as follows:</p><ul><li><p>/ : genotype unphased</p></li>
<li><p>| : genotype phased</p></li>
</ul></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p><!-- %%  ~Describe the value returned -->If do_return = TRUE:</p>
<dl><dt>fix_and_gt</dt>
<dd><p>is a list of two matrices: filtered fixed information (without unnecessary columns) and corresponding genotype section of the VCF</p></dd>
</dl><p>If do_return = FALSE:</p>
<dl><dt>results_directory</dt>
<dd><p>a string with the directory where the results were saved.</p></dd>
</dl></div>
    <div class="section level2">
    <h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a></h2>
    <p>See <a href="https://cran.r-project.org/web/packages/vcfR/vignettes/intro_to_vcfR.html" class="external-link">vcfR package</a> for more information</p>
<p>And also see the "usegalaxy" <a href="https://bit.ly/3oRbjx1" class="external-link">VCFselectsamples</a> tool to pre-filter the data</p>
<p><!-- %% ~put references to the literature/web site here ~ --></p>
    </div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>Tomer Antman</p>
    </div>
    <div class="section level2">
    <h2 id="note">Note<a class="anchor" aria-label="anchor" href="#note"></a></h2>
    <p>Make sure you enter proper file routes (vcf_route) such as:
<!-- % --></p>
<p>1) "somefolder/1011Matrix.gvcf.gz"<br>
 2) "Galaxy4_VCFselectsamples.vcf"</p>

<p>And also proper results route that exist (dir_results) such as:
<!-- % --></p>
<p>1) "somefolder"<br>
 2) "C:/Users/user/Documents"</p>
    </div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p><!-- %% ~~objects to See Also as \code{\link{help}}, ~~~ --></p></div>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span class="va">files_directory</span> <span class="op">&lt;-</span> <span class="fu">Upload_vcf_to_R</span><span class="op">(</span></span>
<span class="r-in">                                  vcf_file <span class="op">=</span> <span class="st">"1011Matrix.gvcf.gz"</span>,</span>
<span class="r-in">                                  dir_results <span class="op">=</span> <span class="st">"C:/Users/user/Documents"</span>,</span>
<span class="r-in">                                  do_return <span class="op">=</span> <span class="cn">FALSE</span></span>
<span class="r-in">                                  <span class="op">)</span></span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>cannot create dir 'C:\Users\user\Documents\100222_16.54_Step1.1-Upload_VCF', reason 'No such file or directory'</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Results directory created:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> C:/Users/user/Documents/100222_16.54_Step1.1-Upload_VCF</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: vcfR</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>    *****       ***   vcfR   ***       *****</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>    This is vcfR 1.12.0 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>      browseVignettes('vcfR') # Documentation</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>      citation('vcfR') # Citation</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>    *****       *****      *****       *****</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>the condition has length &gt; 1 and only the first element will be used</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [[1]]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $src</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "vcf &lt;- read.vcfR(vcf_file, verbose = FALSE)"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> attr(,"class")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "source"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [[2]]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;simpleWarning in file(file, "r"): cannot open file '1011Matrix.gvcf.gz': No such file or directory&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [[3]]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &lt;simpleError in file(file, "r"): cannot open the connection&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 2022-02-10 16:54:28 - getting chromosome info</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: stringr</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: readr</span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in startsWith(vcf@meta, "##contig=&lt;ID=chromosome"):</span> object 'vcf' not found</span>
<span class="r-in"><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">files_directory</span><span class="op">)</span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in h(simpleError(msg, call)):</span> error in evaluating the argument 'x' in selecting a method for function 'print': object 'files_directory' not found</span>
<span class="r-in"></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Tomer Antman.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.9000.</p>
</div>

    </footer></div>

  

  

  </body></html>

